pgmpy/estimators/ExhaustiveSearch.py

```diff
@@ -0,0 +1,185 @@
+#!/usr/bin/env python
+import numpy as np
+import pandas as pd
+import networkx as nx
+from warnings import warn
+from itertools import combinations
+from pgmpy.estimators import StructureEstimator
+from pgmpy.utils.mathext import powerset
+from pgmpy.models import BayesianModel
+
+
+class ExhaustiveSearch(StructureEstimator):
+    def __init__(self, data, scoring_method=None, **kwargs):
+        """
+        Search class for exhaustive searches over all BayesianModels with a given set of variables.
+        Takes a `StructureScore`-Instance as parameter; `estimate` finds the model with maximal score.
+
+        Parameters
+        ----------
+        data: pandas DataFrame object
+            datafame object where each column represents one variable.
+            (If some values in the data are missing the data cells should be set to `numpy.NaN`.
+            Note that pandas converts each column containing `numpy.NaN`s to dtype `float`.)
+
+        scoring_method: Instance of a `StructureScore`-subclass (`K2Score` is used if not set)
+            An instance of `K2Score`, `BdeuScore`, or `BicScore`.
+            This score is optimized during structure estimation by the `estimate`-method.
+
+        state_names: dict (optional)
+            A dict indicating, for each variable, the discrete set of states (or values)
+            that the variable can take. If unspecified, the observed values in the data set
+            are taken to be the only possible states.
+
+        complete_samples_only: bool (optional, default `True`)
+            Specifies how to deal with missing data, if present. If set to `True` all rows
+            that contain `np.Nan` somewhere are ignored. If `False` then, for each variable,
+            every row where neither the variable nor its parents are `np.NaN` is used.
+            This sets the behavior of the `state_count`-method.
+        """
+        if scoring_method is not None:
+            self.scoring_method = scoring_method
+        else:
+            from pgmpy.estimators import K2Score
```

Comment:
@chrisittner Please put all the imports at the top of the file if that isn't leading to a circular import.